lib/Myriad/Registry.pm

```diff
@@ -217,9 +218,20 @@ method receivers_for ($pkg) {
 Reformat a given string to make it combatible with our services
 naming strategy
 
+it takes the following args:
+
+=over 4
+
+=item * name - a String with the original name
+
+=item * namespace - If available the service name will not contain the namespace
+
+=back
+
 =cut
 
-method make_service_name ($name) {
+method make_service_name ($name, $namespace = '') {
+    $name =~ s/$namespace// if $namespace;
```

Comment:
likely needs to be at the start of the string only, and escaping any special characters?

```suggestion
    $name =~ s/^\Q$namespace// if $namespace;
```